File: core/src/main/java/ru/hits/core/domain/dto/user/UserDTO.java
package ru.hits.core.domain.dto.user;

import lombok.Builder;
import lombok.Data;

import java.util.List;
import java.util.UUID;

@Data
@Builder
public class UserDTO {

    private UUID id;
    private String phone;
    private String name;
    private Boolean isBlocked;
    private List<RoleEnum> roles;

}

File: employee-app/HbankPRO/app/src/main/java/com/example/h_bankpro/data/dto/UserDto.kt
package com.example.h_bankpro.data.dto

import com.example.h_bankpro.data.RoleType
import com.example.h_bankpro.domain.model.User
import kotlinx.serialization.Serializable

@Serializable
data class UserDto(
    val id: String,
    val phone: String,
    val name: String,
    val isBlocked: Boolean,
    val roles: List<RoleType>
)

internal fun UserDto.toDomain(): User {
    return User(
        id = id,
        phone = phone,
        name = name,
        isBlocked = isBlocked,
        roles = roles
    )
}